src/openforms/formio/migration_converters.py

```diff
@@ -0,0 +1,51 @@
+"""
+Expose a centralized registry of migration converters.
+
+This registry is used by the data migrations *and* form import. It guarantees that
+component definitions are rewritten to be compatible with the current code.
+"""
+from typing import Protocol
+
+from .typing import Component
+
+
+class Converter(Protocol):
+    def __call__(self, component: Component) -> bool:  # pragma: no cover
+        """
+        Mutate a component in place.
+
+        The component is guaranteed to be of the expected type.
```

Comment:
> I guess since all json that comes from the FormIO always has all keys. It's the same.

that's actually not true, especially not if people add a component in the builder and then edit the JSON. You can specify only the non-default keys and have it work, which is why tests are written with minimal component definitions too :)